x/username/handlers_test.go

```diff
@@ -0,0 +1,268 @@
+package username
+
+import (
+	"context"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/store"
+	"github.com/iov-one/weave/weavetest"
+	"github.com/iov-one/weave/weavetest/assert"
+	"github.com/iov-one/weave/x"
+)
+
+func TestRegisterTokenHandler(t *testing.T) {
+	var (
+		aliceCond = weavetest.NewCondition()
+		bobbyCond = weavetest.NewCondition()
+	)
+
+	cases := map[string]struct {
+		Init           func(t testing.TB, db weave.KVStore)
+		Tx             weave.Tx
+		Auth           x.Authenticator
+		WantCheckErr   *errors.Error
+		WantDeliverErr *errors.Error
+	}{
+		"success": {
+			Tx: &weavetest.Tx{
+				Msg: &RegisterTokenMsg{
+					Username: "bobby@iov",
+					Targets: []BlockchainAddress{
+						{BlockchainID: "bc_1", Address: []byte("addr1")},
+						{BlockchainID: "bc_2", Address: []byte("addr2")},
+					},
+				},
+			},
+			Auth: &weavetest.Auth{Signer: bobbyCond},
+		},
+		"username must be unique": {
+			Tx: &weavetest.Tx{
+				Msg: &RegisterTokenMsg{
+					Username: "alice@iov",
```

Comment:
q: how does this test pass? In your test setup (L86) you have registered `alice*iov` which is not equal to `alice@iov`

Reply:
Tests were wrong. All is fixed now.